java/integrated-windows-authentication/src/main/java/com/esri/arcgisruntime/sample/integratedwindowsauthentication/MainActivity.java

```diff
@@ -85,11 +87,13 @@ protected void onCreate(Bundle savedInstanceState) {
     mRecyclerView = findViewById(R.id.recyclerView);
     mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
 
+    mPortal = new Portal(getString(R.string.arcgis_url));
+
     // set up search public button
     Button searchPublicButton = findViewById(R.id.searchPublicButton);
     searchPublicButton.setOnClickListener(v -> {
       // search the the public ArcGIS portal
-      searchPortal(new Portal(getString(R.string.arcgis_url)));
+      searchPortal(mPortal);
     });
 
     // get reference to load state UI elements
```

Comment:
on line 112 you might want to make that new Portal that is being passed into the `searchPortal` method a class member too, since it will go through the addDone LoadingListener task too.

Reply:
good catch!